### PR TITLE
Начинаем програмировать

### PR DESCRIPTION
**Pull request by dpsnow**
---
:mortar_board: [Начинаем програмировать](https://up.htmlacademy.ru/ecmascript/8/user/574945/tasks/2)

:boom: https://htmlacademy-ecmascript.github.io/574945-taskmanager-8/1/

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -0,0 +1,223 @@
+'use strict';
+
+(function () {
+  const COUNT_CARD = 7;
+  const containerForFilters = document.querySelector(`.main__filter`);
+  const containerForCards = document.querySelector(`.board__tasks`);
+  const listFilters = [
+    {id: `all`, count: getRandomInt(0, 255), isChecked: true},
```
Comment (rkhaslarov) [inline src/main.js]: To be honest, it's not really appropriate to use template strings for simple string without params, probably it'd be better off using single quotes strings here and below as well.

File: src/main.js [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+   *
+   *
+   * @param {string} id
+   * @param {number} count default value = 0
+   * @param {boolean} isChecked default value = false
+   * @return {string} html
+   */
+  const getFillterElement = (id, count = 0, isChecked = false) => {
+    return `<input
+        type="radio"
```
Comment (rkhaslarov) [inline src/main.js]: Please, move don't copy/paste all the html stuff here, probably you could use template tag and place inside index.html

File: src/main.js [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+        ${isChecked === true ? `checked` : ``} />
+      <label for="filter__${id}" class="filter__label">
+        ${id.toUpperCase()}
+        <span class="filter__${id}-count">${count}</span>
+      </label> `;
+  };
+
+  const getCardElement = (colorBarCard, isDeadline = false) => {
+    return `
+    <article class="card card--${colorBarCard} ${isDeadline ? `card--deadline` : `` }">
```
Comment (rkhaslarov) [inline src/main.js]: The same as above

File: src/main.js [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+        </div>
+      </form>
+    </article>`;
+  };
+
+  const renderCards = (count) => {
+    containerForCards.innerHTML = ``;
+    for (let index = 0; index < count; index++) {
+      const htmlCard = getCardElement(getRandomValueArray(listColorBarCard), !!Math.round(Math.random()));
+      containerForCards.insertAdjacentHTML(`beforeend`, htmlCard);
```
Comment (rkhaslarov) [inline src/main.js]: I'd prefer to use template cloning and document fragments

File: src/main.js [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+    }
+  };
+
+  listFilters.forEach((filterParams) => {
+    containerForFilters.insertAdjacentHTML(`beforeend`, getFillterElement(
```
Comment (rkhaslarov) [inline src/main.js]: The same as above + you could use object and destructuring syntax for function args
function getFillterElement({ id, count, isChecked  }) { ... }
and call it getFillterElement(filterParams)

File: src/main.js (hunk 1 above)
Comment (dpsnow) [reply src/main.js]: Unfortunately I get errors from eslint and "The build failed"

**Pull request merged**